### PR TITLE
Fix #7092 - Catch errors in GroundHeatExchanger:HorizontalTrench when pipe inner/outer diameters don't match

### LINKED ISSUE #7092: Ground heat exchanger divide by zero if user inputs equal pipe inner and outer diameters
Issue overview
--------------
User file with equal values for `GroundHeatExchanger:HorizontalTrench` "Pipe Inner Diameter" and "Pipe Outer Diameter" results in a divide-by-zero [here](https://github.com/NREL/EnergyPlus/blob/ae4c2ce0d208dff6982b852c40fccb78aa78c552/src/EnergyPlus/PlantPipingSystemsManager.cc#L6428-L6430) because [`XY_CrossSectArea`](https://github.com/NREL/EnergyPlus/blob/ae4c2ce0d208dff6982b852c40fccb78aa78c552/src/EnergyPlus/PlantPipingSystemsManager.hh#L257-L261) returns zero. In the release build, this propogates NANs until this familiar and confusing fatal error occurs:

```
   ** Severe  ** DualSetPointWithDeadBand: Unanticipated combination of heating and cooling loads - report to EnergyPlus Development Team
   **   ~~~   ** occurs in Zone=ZONE_1_14 During Warmup, Environment=WASHINGTON DC DULLES INTL AR ANN CLG 1% CONDNS DB=>MWB, at Simulation time=07/21 23:00 - 24:00
   **   ~~~   ** LoadToHeatingSetPoint=NAN, LoadToCoolingSetPoint=NAN
   **   ~~~   ** Zone Heating Set-point=21.11
   **   ~~~   ** Zone Cooling Set-point=22.78
   **   ~~~   ** Zone TempDepZnLd=NAN
   **   ~~~   ** Zone TempIndZnLd=NAN
   **   ~~~   ** Zone ThermostatSetPoint=22.78
   **  Fatal  ** Program terminates due to above conditions.
```

Input processing for `GroundHeatExchanger:HorizontalTrench` should probably require a minimal positive value for Outer Diameter - Inner Diameter, otherwise throw a severe/fatal error.

`GroundHeatExchanger:Pond` also has inputs for inner and outer diameter, so it may have the same issue.

Other ground heat exchangers have a diameter plus thickness which must be >0, or just one diameter(?), so they don't have the same problem.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Helpdesk ticket number 12268.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Fix #7092 - Catch errors in GroundHeatExchanger:HorizontalTrench when pipe outer diameter isn't stritcly greater than inner diameter 

Changelog:
* Corrected small issue in existing reporting for the other object that has inner&outer diameters (`PipingSystem:Underground:PipeCircuit`)
* Added catching error + reporting for `GroundHeatExchanger:HorizontalTrench`
* Wrote a unit test to test both `PipingSystem:Underground:PipeCircuit` and `GroundHeatExchanger:HorizontalTrench`
* Modified `NodeInputManager::GetNodeListsInput(bool &ErrorsFound)` to not set `ErrorsFound` to false (found this problem while testing for `PipingSystem:Underground:PipeCircuit`)

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by jmarrec** [comment]
I have some failing tests that I'll need to double check now, all are due to the change in `NodeInputManager::GetNodeListsInput(bool &ErrorsFound)`. It had previously been setting ErrorsFound to false at some point and preventing a fatal.

File: src/EnergyPlus/NodeInputManager.cc [hunk 1]
```diff
@@ -781,16 +781,17 @@ namespace NodeInputManager {
...
             }
         }
 
         cAlphas.deallocate();
         rNumbers.deallocate();
 
-        if (ErrorsFound) {
+        if (localErrorsFound) {
             ShowFatalError(RoutineName + CurrentModuleObject + ": Error getting input - causes termination.");
+            ErrorsFound = true;
```
Comment (Myoldmopar) [inline src/EnergyPlus/NodeInputManager.cc:794]: If you throw fatal in the line above this, then this line is not reachable.

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 2]
```diff
@@ -2004,6 +2004,15 @@ namespace PlantPipingSystemsManager {
...
             HGHX(HorizontalGHXCtr).PipeID = rNumericArgs(5);
             HGHX(HorizontalGHXCtr).PipeOD = rNumericArgs(6);
+
+            // Issue a severe if Inner >= Outer diameter
+            if (HGHX(HorizontalGHXCtr).PipeID >= HGHX(HorizontalGHXCtr).PipeOD) {
+                ShowSevereError(RoutineName + ": " + ObjName_HorizTrench + "=\"" + cAlphaArgs(1) + "\" has invalid pipe diameters.");
+                ShowContinueError("Outer diameter [" + General::TrimSigDigits(HGHX(HorizontalGHXCtr).PipeOD, 3)
+                        + "] must be greater than inner diameter [" + General::TrimSigDigits(HGHX(HorizontalGHXCtr).PipeID, 3) + "].");
+                ErrorsFound = true;
+            }
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantPipingSystemsManager.cc:2014]: :+1:

File: tst/EnergyPlus/unit/PlantPipingSystemsManager.unit.cc [hunk 3]
```diff
@@ -1819,3 +1820,185 @@ TEST_F(EnergyPlusFixture, PipingSystemFullSimulation)
...
     PlantPipingSystemsManager::SimulateGroundDomains(false);
 }
+
+/*
+ * Asserts that all objects that defined both an inside and an outside diameters have correct input (inside < outside)
+ * - PipingSystem:Underground:PipeCircuit
+ * - GroundHeatExchanger:HorizontalTrench
+ */
+
+TEST_F(EnergyPlusFixture, PipingSystem_Check_Correct_Pipe_Diameters) {
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/PlantPipingSystemsManager.unit.cc:1830]: :+1: :+1: :+1:

**Review by Myoldmopar — COMMENTED** [review]
We should finalize what to do about that errors found variable in NodeInputManager, since if it still calls ShowFatalError then the errors found variable changes appear to be unused.

**Comment by jmarrec** [comment]
@myoldmopar I cherry picked the ErrorsFound commit on the pr aiming to fix #7147 that I'll open on Monday I think, so we could decide it there instead. I can revert this commit on this PR if you want

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
This is good to go.  CI is very happy, and I am glad that it isolates the errors-found change to just the context of this PR for now.  Thanks!

**Pull request merged**